tests/test_formatting.py

```diff
@@ -859,3 +860,35 @@ def test_format_template_no_module(sourceless_class: type[Any]) -> None:
 value = 0
 """.strip()
     )
+
+
+class Verbosity(Enum):
+    QUIET = auto()
+    NORMAL = auto()
+    DETAILED = auto()
+
+
+class IntVerbosity(IntEnum):
+    QUIET = 0
+    NORMAL = 1
+    DETAILED = 2
+
+
+def test_format_with_enums() -> None:
+    @dataclass
+    class Log:
+        message: str
+        verbosity: Verbosity
+        verbosity_level: IntVerbosity
+
+    log = Log("Hello, World", Verbosity.DETAILED, IntVerbosity.DETAILED)
+
+    template = "\n".join(Binder(log).format_toml())
+
+    assert template == (
+        """
+message = 'Hello, World'
+verbosity = 'detailed'
+verbosity-level = 2
```

Comment:
The actual test output is `verbosity-level = IntVerbosity.DETAILED` in my development environment. I think the expected output in your test case is correct, so then the template formatting implementation must be incorrect.

Can you reproduce this on Python 3.11/3.12 or is it possibly a problem in my 3.10 `ReprEnum` workaround?

Reply:
On Python 3.11/3.12 this works as intended, however I was able to reproduce this in Python  3.10

Reply:
The interesting part is why does this work in Python 3.11/3.12? :)

Because IntEnum is of type int and ReprEnum and the first check should probably fail this already? Fix was quite easy we just need to check if its an enum first